src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -138,6 +138,36 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                                         printPackages = false;
                                     }
                                 }
+                                // Handle --deprecated
+                                else if (listPackageArgs.IncludeDeprecated)
+                                {
+                                    await GetDeprecationInfoAsync(packages, listPackageArgs);
+
+                                    var noPackagesLeft = true;
```

Comment:
suggestion: 
how about a better name,
deprecatedPackagesFound and initialized to true?  

This feels weird.

Reply:
now I see this is a big method, so there's already a package with a similar name, so feel free to make a judgment call here. 

Not a fan of the whole print packages utility flow :)

Reply:
Completely killed the boolean. Refactored a little.